src/lib.rs

```diff
@@ -941,6 +941,62 @@ pub mod invariant {
         fn get_fee_tiers(&self) -> Vec<FeeTier> {
             self.fee_tiers.get_all()
         }
+
+        #[ink(message)]
+        fn get_position_ticks(&self, owner: AccountId, offset: u16) -> Vec<PositionTick> {
+            let positions = self.positions.get_all(owner);
```

Comment:
You can significantly optimize this implementation if you load only the required position states from the map. Currently, you load all in regardless of the offset